internal/sum/sum.go

```diff
@@ -0,0 +1,279 @@
+package sum
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	ex "github.com/bmc-toolbox/bmclib/v2/internal/executor"
+
+	"github.com/bmc-toolbox/common"
+	"github.com/bmc-toolbox/common/config"
+	"github.com/go-logr/logr"
+)
+
+// Sum is a sum command executor object
+type Sum struct {
+	Executor ex.Executor
+	SumPath  string
+	Log      logr.Logger
+	Host     string
+	Username string
+	Password string
+}
+
+// Option for setting optional Client values
+type Option func(*Sum)
+
+func WithSumPath(sumPath string) Option {
+	return func(c *Sum) {
+		c.SumPath = sumPath
+	}
+}
+
+func WithLogger(log logr.Logger) Option {
+	return func(c *Sum) {
+		c.Log = log
+	}
+}
+
+func New(host, user, pass string, opts ...Option) (*Sum, error) {
+	sum := &Sum{
+		Host:     host,
+		Username: user,
+		Password: pass,
+		Log:      logr.Discard(),
+	}
+
+	for _, opt := range opts {
+		opt(sum)
+	}
+
+	var err error
+
+	if sum.SumPath == "" {
+		sum.SumPath, err = exec.LookPath("sum")
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		if _, err = os.Stat(sum.SumPath); err != nil {
+			return nil, err
+		}
+	}
+
+	e := ex.NewExecutor(sum.SumPath)
+	e.SetEnv([]string{"LC_ALL=C.UTF-8"})
+	// e.SetQuiet()
+	sum.Executor = e
+
+	return sum, nil
+}
+
+// Return a Fake sum executor for tests
+func NewFakeSum(r io.Reader) (*Sum, error) {
```

Comment:
Any reason this can't live in a test package only?

Reply:
Nope, and you're correct it's not used anywhere.   Bye bye.